src/controls/PrimitiveArrayControl.tsx

```diff
@@ -0,0 +1,148 @@
+import {
+  Helpers,
+  ArrayLayoutProps,
+  composePaths,
+  createDefaultValue,
+  findUISchema,
+} from "@jsonforms/core"
+import {
+  JsonFormsDispatch,
+  withJsonFormsArrayControlProps,
+} from "@jsonforms/react"
+import { Form, Button, Col, Row } from "antd"
+import type { Rule } from "antd/es/form"
+import { useEffect, useMemo } from "react"
+import { ArrayControlOptions } from "../ui-schema"
+import { usePreviousValue } from "../common/usePrevious"
+import React from "react"
+
+export function PrimitiveArrayControl({
+  data,
+  enabled,
+  path,
+  schema,
+  uischema,
+  addItem,
+  removeItems,
+  renderers,
+  cells,
+  visible,
+  rootSchema,
+  uischemas,
+  required,
+}: ArrayLayoutProps) {
+  const foundUISchema = useMemo(
+    () =>
+      findUISchema(
+        uischemas ?? [],
+        schema,
+        uischema.scope,
+        path,
+        undefined,
+        uischema,
+        rootSchema,
+      ),
+    [uischemas, schema, path, uischema, rootSchema],
+  )
+
+  const addItemToList = useMemo(
+    () => addItem(path, createDefaultValue(schema, rootSchema)),
+    [addItem, path, schema, rootSchema],
+  )
+
+  const prevDataValue = usePreviousValue(data)
+  useEffect(() => {
+    if (data === undefined && prevDataValue === null) {
+      addItemToList()
+    }
+  })
```

Comment:
Same suggestion here about useEffect without dep arrays

Reply:
I actually was able to add the deps array here. It's only an issue with `ObjectArrayControl`.